components/builder-api-client/src/lib.rs

```diff
@@ -1145,35 +1144,21 @@ impl Client {
             .get::<XFileName>()
             .expect("XFileName missing from response")
             .to_string();
-        let tmp_file_path = dst_path.join(format!(
-            "{}.tmp-{}",
-            file_name,
-            thread_rng()
-                .sample_iter(&Alphanumeric)
-                .take(8)
-                .collect::<String>()
-        ));
         let dst_file_path = dst_path.join(file_name);
-        debug!("Writing to {}", &tmp_file_path.display());
-        let mut f = File::create(&tmp_file_path)?;
+        let mut w = AtomicWriter::new(&dst_file_path)?;
         match progress {
-            Some(mut progress) => {
+            Some(ref mut progress) => {
                 let size: u64 = res
                     .headers
                     .get::<hyper::header::ContentLength>()
                     .map_or(0, |v| **v);
                 progress.size(size);
-                let mut writer = BroadcastWriter::new(&mut f, progress);
-                io::copy(&mut res, &mut writer).map_err(Error::BadResponseBody)?
```

Comment:
Why no more `map_err` call?

Reply:
So, I can definitely add these back, but they kinda feel potentially wrong. Can't the error be on the read or the write side here?